pages/reader/epub.vue

```diff
@@ -612,6 +429,50 @@ async function loadEPub() {
     percentage.value = book.locations.percentageFromCfi(location.start.cfi)
     currentCfi.value = location.start.cfi
   })
+
+  const ttsSegments = await extractTTSSegments(book)
+  setTTSSegments(ttsSegments)
+}
+
+async function extractTTSSegments(book: ePub.Book) {
+  const ttsSegments: TTSSegment[] = []
+  const sections: Section[] = []
+  let globalIndex = 0
+
+  book.spine.each((section: Section) => {
+    sections.push(section)
+  })
+
+  for (const section of sections) {
+    try {
+      await section.load(book.load.bind(book))
+
+      const doc = section.document
+      if (!doc) {
+        console.warn(`No document found for section ${section.href}`)
+        continue
+      }
+
+      const elements = Array.from(doc.querySelectorAll('p, h1, h2, h3, h4, h5, h6'))
+        .filter(el => !!el.textContent?.trim())
+
+      elements.forEach((el, elIndex) => {
+        const text = el.textContent?.trim() || ''
+        ttsSegments.push(
+          ...splitTextIntoSegments(text).map((segment, segIndex) => ({
+            text: segment,
+            id: `${section.index}-${elIndex}-${segIndex}`,
+            index: globalIndex++,
```

Comment:
Do we really need to store this index if we already push all segments as array in `setTTSSegments`? Just get the current array index in tts composable maybe, no need to store as a extra key value.

On other hand maybe we want to store `section.href` and maybe `chapter`

Reply:
[🎨 Add href & chapter to segments](https://github.com/likecoin/3ook-com/pull/179/commits/b4d1f0498c3a4f7e8f0639738fa22fa5f718d9cc)
[🎨 Initialize TTS start index](https://github.com/likecoin/3ook-com/pull/179/commits/3b69056e79e76784291050ab0674da8b28e5b788)